### DISCUSSION TITLE
dynamic windowing - how to show multiple windows created at runtime?

### DISCUSSION THREAD

**Discussion by nyvs** [emilk/egui#1354] [Q&A]
Hi, in my Panel I have a Scrollable list with a grid, with items numbered through from 1 to n.
each item has a details button. Now I would like to open a new window for an arbitrary item on click.
My first solution won't work because of borrow checking, how can I solve this without interior mutability?
```rust
pub struct DetailList {
	search: String,
	windows: Vec<(bool, DetailWindow)>
}

//in show(), this wont work:
for window in self.windows.iter_mut() {
	egui::Window::new("detail window")
	.open(&mut window.0)
	.resizable(true)
	.show(ctx, |ui| { //closure requires unique access to `window` but it is already borrowed
		use super::View as _;
		window.1.ui(ui);
	});
}
```

I think I'm thinking too complicated haha, many thanks

**Comment by emilk** [comment]
This is not really the place to ask questions about the borrow checker, but in this case the solution is rather simple:

`for (open, window) in &mut self.windows {`
